### PR TITLE
modal enhancements for create and update

### PR DESCRIPTION
**Pull request by pooja-koul-incontact**

### CODE REVIEW

File: src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css [hunk 1]
```diff
@@ -4,6 +4,7 @@
   position: absolute;
   font-family: 'OpenSans';
   z-index: 12;
+  margin-left: -255px;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css]: avoid CSS in negative value

File: src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css [hunk 2]
```diff
@@ -29,17 +30,31 @@
 }
 .createRuleModalContainer .modal-content{
   border-radius: 0;
-  min-width: 175%;
+  min-width: 224%;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css]: should not be more than 100%

File: src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css [hunk 3]
```diff
@@ -311,10 +326,20 @@
 }
 .addUsersToRuleContainer .modal-content{
   border-radius: 0;
-  min-width: 175%;
+  min-width: 224%;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-create-rule-modal/bu-create-rule-modal.component.css]: should not be more than 100%. Not a valid value.

File: src/app/bu/bu-rule-mapping/bu-display-rule-mapping/bu-rule-mapping.component.spec.ts [hunk 4]
```diff
@@ -52,12 +53,16 @@ describe('BuRuleMappingComponent', () => {
     expect(component.onClickDismissModal).toBeTruthy();
     expect(!component.showCreateRuleModal);
   });
+  it('should call init method', () => {
+    spyOn(component, 'ngOnInit');
+    component.ngOnInit();
+    expect(component.ngOnInit).toBeTruthy();
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-display-rule-mapping/bu-rule-mapping.component.spec.ts]: should be similar to expect(component.ngOnInit).toHaveBeenCalled();

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 5]
```diff
@@ -24,20 +24,35 @@
 }
 .modal-dialog {
   position: absolute;
-  margin-top: -780px;
+  margin-top: -800px;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: Provided negative values. Please avoid using negative in CSS

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 6]
```diff
@@ -24,20 +24,35 @@
...
+  margin-left: -255px;
   z-index: 12;
   font-family: 'OpenSans';
 }
 .updateRuleModalContainer .modal-content {
-  min-width: 175%;;
+  min-width: 224%;;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: should not exceed 100%.

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 7]
```diff
@@ -24,20 +24,35 @@
...
-  .updateRuleModal {
-    width: 45%;
-    margin-left: 30px;
+  .updateRuleModalContainer .updateRuleModal {
+    width: 850px;
   }
-  .updateRuleModalBody {
-    height: 440px;
+  .updateRuleModalContainer .updateRuleModalBody {
+    min-height: 630px;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: generally we don't give min-height. It should actually be depends on the elements inside it,

File: src/app/bu/bu-rule-mapping/bu-display-rule-mapping/bu-rule-mapping.component.css [hunk 8]
```diff
@@ -17,7 +17,6 @@
 }
 .ruleMappingContainer .header-buttons-container {
   margin-top: 15px;
-  margin-right: 45px;
   margin-bottom: -50px;
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-display-rule-mapping/bu-rule-mapping.component.css]: give some +ve number

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 9]
```diff
@@ -380,15 +395,29 @@
   margin-top: -50px;
 }
 .updateAddUsersToRuleContainer .modal-content {
-  min-width: 175%;;
+  min-width: 224%;;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: it should not exceed 100%

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 10]
```diff
@@ -380,15 +395,29 @@
...
+  }
+}
+@media (min-width: 1800px) {
+  .updateAddUsersToRuleContainer .modal-content {
+    height: 870px;
+  }
+}
+@media (min-width: 800px) {
+  .updateAddUsersToRuleContainer .updateRuleModalBody {
+    min-height: 385px;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: generally we don't give min-height. It should actually be depends on the elements inside it,

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css [hunk 11]
```diff
@@ -380,15 +395,29 @@
...
-  .updateRuleModal {
-    width: 45%;
-    margin-left: 30px;
+  .updateAddUsersToRuleContainer .updateRuleModal {
+    width: 850px;
   }
-  .updateRuleModalBody {
-    height: 440px;
+  .updateAddUsersToRuleContainer .updateRuleModalBody {
+    min-height: 695px;
```
Comment (akash-nema-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: same goes here

File: src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-creation-date/bu-rule-mapping-creation-date.component.ts [hunk 12]
```diff
@@ -8,7 +9,7 @@ import {Router} from '@angular/router';
 })
 export class BuRuleMappingCreationDateComponent implements OnInit {
 
-  constructor(private router: Router) { }
+  constructor(private router: Router, private userMappingDataService: BuUserMappingDataService) { }
```
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-rule-mapping-grid/bu-rule-mapping-creation-date/bu-rule-mapping-creation-date.component.ts]: do you need userMappingDataService , if not remove it

File: src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css (hunk 9 above)
Comment (Susmith-roy-incontact) [inline src/app/bu/bu-rule-mapping/bu-update-rule-mapping/bu-update-rule-mapping.component.css]: if possible use pixel instead of %, remove ;

**Pull request merged**